pkg/remote/cache/cache_test.go

```diff
@@ -69,4 +69,39 @@ func TestCache(t *testing.T) {
 		}
 		assert.Equal(t, 5, cache.Len())
 	})
+
+	t.Run("should ignore keys when capacity is 0", func(t *testing.T) {
+		keys := []struct {
+			key           string
+			value         interface{}
+			overridden    bool
+			expectedValue interface{}
+		}{
+			{
+				"test",
+				[]string{"slice"},
+				false,
+				nil,
+			},
+			{
+				"test",
+				[]string{},
+				false,
+				nil,
+			},
+			{
+				"test2",
+				[]resource.FakeResource{},
+				false,
+				nil,
+			},
+		}
+		cache := New(0)
+
+		for _, k := range keys {
+			assert.Equal(t, k.overridden, cache.Put(k.key, k.value))
+			assert.Equal(t, k.expectedValue, cache.Get(k.key))
```

Comment:
No need to have an `overriden` bool in your struct since with a cache of cap 0 it must be everytime `false`. Same for `expectedValue`.